packages/next/client/image.tsx

```diff
@@ -261,32 +261,27 @@ function defaultImageLoader(loaderProps: ImageLoaderProps) {
 
 // See https://stackoverflow.com/q/39777833/266535 for why we use this ref
 // handler instead of the img's onLoad attribute.
-// 1500ms delay in removing placeholder is to prevent flash of white between
-// image load and image render.
 function removePlaceholder(
   element: HTMLImageElement | null,
   placeholder: PlaceholderValue
 ) {
   if (placeholder === 'blur' && element) {
-    if (element.complete && !element.src.startsWith('data:')) {
+    const handleLoad = () => {
+      if (!element.src.startsWith('data:')) {
+        ;(element.decode() || Promise.resolve()).then(() => {
```

Comment:
This may not work in older browsers, as `element.decode` may be `undefined`. Please make sure to check for `element.decode != null`.

Reply:
```suggestion
        (element.decode || Promise.resolve)().then(() => {
```

Reply:
Nice catch!

Reply:
I fixed it in 38c65a8